proxy/dial.go

```
@@ -0,0 +1,55 @@
+	if cfg.Proxy.Url == "" {
+		transport.Proxy = http.ProxyFromEnvironment
+		return
+	}
+
+	proxyInfo, err := url.Parse(cfg.Proxy.Url)
+	if err == nil {
+		if strings.HasPrefix(cfg.Proxy.Url, "http") {
+			transport.Proxy = http.ProxyURL(proxyInfo)
+		} else {
+			proxyDialer := newProxyDial(cfg.Proxy.Url)
+			transport.Dial = proxyDialer.Dial
```

golangci-lint (1.62.2)

SA1019: transport.Dial has been deprecated since Go 1.7: Use DialContext instead, which allows the transport to cancel dials as soon as they are no longer needed. If both are set, DialContext takes priority. 

(staticcheck)